src/cosmos/cosmosUtils.ts

```diff
@@ -0,0 +1,24 @@
+import { HttpEndpoint } from '@cosmjs/stargate'
+
+import {
+  asCosmosInitOptions,
+  CosmosInitOptions,
+  CosmosNetworkInfo
+} from './cosmosTypes'
+
+export const rpcWithApiKey = (
+  networkInfo: CosmosNetworkInfo,
+  initOptions: CosmosInitOptions
+): HttpEndpoint => {
+  const apiKeys = asCosmosInitOptions(initOptions)
+  const endpoint = networkInfo.rpcNode
+  const headers = {}
+  for (const [key, value] of Object.entries(endpoint.headers)) {
+    if (value in apiKeys) {
+      // @ts-expect-error
```

Comment:
remove // @ts-expect-error and change line 13 to

`  const apiKeys = asCosmosInitOptions(initOptions) as { [key: string]: string }`